cmd/end2endtest/helpers.go

```diff
@@ -165,84 +161,51 @@ func (t *e2eElection) waitUntilElectionStarts(electionID types.HexBytes) (*vapi.
 	return election, nil
 }
 
-func (t *e2eElection) generateProofs(root types.HexBytes, isAnonymousVoting bool, csp *ethereum.SignKeys) map[string]*apiclient.CensusProof {
-	type voterProof struct {
-		proof   *apiclient.CensusProof
-		address string
-	}
-	proofs := make(map[string]*apiclient.CensusProof, len(t.voterAccounts))
-	proofCh := make(chan *voterProof)
-	stopProofs := make(chan bool)
-	go func() {
-		for {
-			select {
-			case p := <-proofCh:
-				proofs[p.address] = p.proof
-			case <-stopProofs:
-				return
-			}
-		}
-	}()
+func (t *e2eElection) generateProofs(csp *ethereum.SignKeys, voterAccts []*ethereum.SignKeys) {
+	var wg sync.WaitGroup
+	var vcount int32
 
-	apiClientMtx := &sync.Mutex{}
 	addNaccounts := func(accounts []*ethereum.SignKeys, wg *sync.WaitGroup) {
 		defer wg.Done()
 		log.Infof("generating %d voting proofs", len(accounts))
 		for _, acc := range accounts {
 			voterKey := acc.Address().Bytes()
 
-			if isAnonymousVoting {
-				apiClientMtx.Lock()
+			if t.election.VoteMode.Anonymous {
 				privKey := acc.PrivateKey()
 				if err := t.api.SetAccount(privKey.String()); err != nil {
-					apiClientMtx.Unlock()
 					log.Fatal(err)
-					return
 				}
 				voterKey = t.api.MyZkAddress().Bytes()
 			}
 
-			var pr *apiclient.CensusProof
-			var err error
-			if csp != nil {
-				pr, err = cspGenProof(t.election.ElectionID, voterKey, csp)
-			} else {
-				pr, err = t.api.CensusGenProof(root, voterKey)
-				if isAnonymousVoting {
-					apiClientMtx.Unlock()
+			pr, err := func() (*apiclient.CensusProof, error) {
+				if csp != nil {
+					return cspGenProof(t.election.ElectionID, voterKey, csp)
 				}
-			}
-
+				return t.api.CensusGenProof(t.election.Census.CensusRoot, voterKey)
+			}()
 			if err != nil {
 				log.Fatal(err)
 			}
 
-			if !isAnonymousVoting {
-				pr.KeyType = models.ProofArbo_ADDRESS
-			}
-			proofCh <- &voterProof{
-				proof:   pr,
-				address: acc.Address().Hex(),
-			}
+			t.voters.Store(acc.Public, acctProof{account: acc, proof: pr})
```

Comment:
I'm starting to think - why do you need a `sync.Map` at all? presumably you always set up the election before running it, so the voters map is only written to first, and then later only read from. A regular Go map is fine for that. You only need a `sync.Map` if the writes are concurrent with other reads or writes, but if that were to happen, the test would likely be wrong - you would be starting the election before the setup has finished, for example.

Reply:
The idea of use `sync.Map` comes from a previous suggestion in a [PR comment](https://github.com/vocdoni/vocdoni-node/pull/997#discussion_r1234184159) for the vote proof refactor. I was not sure about using it due the documentation mention 2 cases: 
```
The Map type is optimized for two common use cases: (1) when the entry for a given
key is only ever written once but read many times, as in caches that only grow,
 or (2) when multiple goroutines read, write, and overwrite entries for disjoint
 sets of keys.
```
But, I wanted to confirm if using `sync.Map` could be a option or not, so I open a PR as proposal.

Reply:
the whole anon func is called concurrently (as long as pcount > 1, default 4) 

`go addNaccounts(voterAccts[i:end], &wg)`